backend/views/views.py

```diff
@@ -34,12 +37,40 @@ def admin():
     return render_template('admin_page.html')
 
 
-@app.route('/userpage')
+@app.route('/userpage', methods=['GET', 'POST'])
 @admin_permissions
 def user_page():
-    users = db.session.query(User, Role).filter(
-        User.role_id == Role.id).order_by(User.id).all()
-    return render_template('user_page.html', users=users)
+    form = SearchForm(request.args, csrf_enabled=False)
+    if  form.validate():
+        key = int(request.args.get('field_by'))
+        search_string = str(request.args.get('search'))
+        search_paremeters = []
+        condition_list = []
+        for one_string in search_string.split():
+            if len(one_string) >= MIN_SEARCH_STR:
+                search_paremeters.append(''.join(["%", one_string, "%"]))
```

Comment:
`''.join(["%", one_string, "%"])` is a bit ugly, but `'%%%s%%' % one_string` is confusing too. So I'd recommend to use `'%{}%'.format(one_string)` here.